snuba/query/parser/expressions.py

```diff
@@ -159,19 +199,29 @@ def parse_aggregation(
 
     columns_expr = [parse_expression(column) for column in columns if column]
 
-    if isinstance(parsed_expression, str):
-        # Simple aggregation with snuba syntax ["count", ["c1", "c2"]]
-        return FunctionCall(alias, parsed_expression, tuple(columns_expr))
-    elif (
-        # Simple Clickhouse expression with no snuba syntax
-        # ["ifNull(count(somthing), something)", None, None]
-        isinstance(parsed_expression, (FunctionCall, CurriedFunctionCall))
-        and not columns_expr
-    ):
-        return replace(parsed_expression, alias=alias)
-    elif isinstance(parsed_expression, FunctionCall) and columns_expr:
-        # Mix of clickhouse syntax and snuba syntax that generates a CurriedFunction
-        # ["f(a)", "b", None]
-        return CurriedFunctionCall(alias, parsed_expression, tuple(columns_expr),)
+    matched = re.fullmatch("[a-zA-Z_][a-zA-Z0-9_]*", aggregation_function)
```

Comment:
Let's not duplicate this regex if possible since it's the same one as function name. Can we reference it from minimal_clickhouse_grammar instead?

Reply:
To avoid duplicates, we declared the function_name regex outside of the grammar, then used it in the grammar and in the regex matcher. Let me know what you think!